File: js/problem14.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) {
        return '';
    } else if (strs.length === 1) {
        return strs[0];
    }
    
    let i = 0;
    let flag = true;
    while (i < strs[0].length) {
        let reference = strs[0][i];
        for (let j = 1; j < strs.length; j++) {
            if (i >= strs[j].length) {
                flag = false;
                break;
            }
            
            if (strs[j][i] !== reference) {
                flag = false;
                break;
            }
        }
        
        if (!flag) {
            break;
        }
        i++;
    }
    
    return strs[0].slice(0, i);
};

File: leetcodeJava/src/main/java/Problem14.java
// 14. Longest Common Prefix

class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        } else if (strs.length == 1) {
            return strs[0];
        }
        
        int res = 0;
        
        while (res < strs[0].length()) {
            boolean flag = true;
            for (int i = 1; i < strs.length; i++) {
                if ((res >= strs[i].length()) || (strs[0].charAt(res) != strs[i].charAt(res))) {
                    flag = false;
                    break;
                }
            }
            
            if (!flag) {
                break;
            }
            res++;
        }
        
        return strs[0].substring(0, res);
    }
}